jwst/associations/lib/dms_base.py

```diff
@@ -1141,6 +1141,43 @@ def nrslamp_valid_detector(item):
     return False
 
 
+def nrs_nod_background_overlap(science_item, background_item):
+    """
+    Check that a candidate background nod will not overlap with science.
+
+    Currently, this function only returns True for NIRSpec
+    fixed slit exposures taken with slit S1600A1 in a 5-point
+    nod pattern, when the background candidate is in the
+    closest nod to the science.
+    """
+    try:
+        _, exptype = item_getattr(science_item, ['exp_type'])
+        _, slit = item_getattr(science_item, ['fxd_slit'])
+        _, numdthpt = item_getattr(science_item, ['numdthpt'])
+        _, sci_num = item_getattr(science_item, ['patt_num'])
+        _, bkg_num = item_getattr(background_item, ['patt_num'])
+    except KeyError:
+        return False
+
+    try:
+        numdthpt = int(numdthpt)
+        sci_num = int(sci_num)
+        bkg_num = int(bkg_num)
+    except ValueError:
+        return False
+
+    # Background is currently only expected to overlap severely for
+    # 5 point dithers with fixed slit S1600A1.  Only the nearest
+    # dithers to the science observation need to be excluded.
+    if (exptype == 'nrs_fixedslit'
+            and slit == 's1600a1'
+            and numdthpt == 5
+            and abs(sci_num - bkg_num) <= 1):
```

Comment:
Patt_num increments with each sub-pixel dither as well as the primary dithers, and the order of the dithers is that it will step through each sub-pixel dither before stepping to the next primary nod position (at least for spectral sub-pixel dithers), so I think this might not work as expected for sub-pixel dithered data.  I don't think the primary nod-number is saved as a header keyword, but for some other fixed slit cases I had grabbed the number of sub-pixel dithers ("subpxpts") and calculated nod numbers with:
`nod_num = ((pattnum - 1) // subpxpts) + 1`, which might work here (I think subpxpts should be accessible).  And then I think you could use the same kind of rejection for any nod_num that are <= 1 apart.

Reply:
Sounds good -- looking at the sequence of dither positions Charles attached to the ticket, I think this ought to work.